src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double RESIZE_COUNT = 1.5;
+    private Object[] elementData;
+    private int defaultSize;
+    private int count;
+
+    public ArrayList() {
+        defaultSize = 10;
+        elementData = new Object[defaultSize];
+        count = 0;
+    }
+
+    private boolean rangeCheckForCount(int index) {
+        if (index >= count || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound of array.");
```

Comment:
You can make message more informative by adding index value to it, like it made in standard java library (e.g. "Index 2 out of bounds for length 2").